### PR TITLE
[receiver/kubeletstats] Add new memory utilization metrics

### PR DESCRIPTION
**Pull request by TylerHelmuth**
**Description:**
Adds new `k8s.pod.memory.utilization` and `container.memory.utilization` metrics that represent the ratio of memory used vs limits set.

The metrics are only emitted for pods/containers that have defined resource limits. It takes advantage of the pod metadata to acquire the container limits. A pod limit is computed as the sum of all the container limits and if any container limit is zero or undefined the pod limit is also considered undefined.

**Link to tracking Issue:**
Related to https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/24905

**Testing:**
Unit tests and local testing

### CODE REVIEW

**Comment by dmitryax** [comment]
A good comment was brought in the issue: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/24905#issuecomment-1684122962. Let's discuss that first.

**Comment by TylerHelmuth** [comment]
@dmitryax if you feel good about the `memory_limit` and `memory_request` naming I can update this PR.

**Comment by dmitryax** [comment]
> if you feel good about the memory_limit and memory_request naming I can update this PR.

SGTM

**Comment by TylerHelmuth** [comment]
@dmitryax updated the names and implemented `requests` logic.

**Comment by TylerHelmuth** [comment]
@dmitryax this is ready for a proper review.

File: receiver/kubeletstatsreceiver/internal/kubelet/mem.go [hunk 1]
```diff
@@ -21,4 +23,15 @@ func addMemoryMetrics(mb *metadata.MetricsBuilder, memoryMetrics metadata.Memory
 	recordIntDataPoint(mb, memoryMetrics.WorkingSet, s.WorkingSetBytes, currentTime)
 	recordIntDataPoint(mb, memoryMetrics.PageFaults, s.PageFaults, currentTime)
 	recordIntDataPoint(mb, memoryMetrics.MajorPageFaults, s.MajorPageFaults, currentTime)
+
+	if s.UsageBytes != nil {
+		if r.memoryLimit > 0 {
+			value := math.Min(float64(*s.UsageBytes)/float64(r.memoryLimit), 1)
```
Comment (dmitryax) [inline receiver/kubeletstatsreceiver/internal/kubelet/mem.go:29]: Why do we need to cap it at 1? Can't the usage go above the limit right before being killed?

Comment (TylerHelmuth) [reply receiver/kubeletstatsreceiver/internal/kubelet/mem.go:29]: Yes. It comes down to whether we want to represent both limit and request as soft limits. Our earlier discussions had limit being a hard limit and request being a soft limit, but they could both be treated as soft limits if we want.

**Comment by TylerHelmuth** [comment]
@dmitryax ready for review

**Review by dmitryax — APPROVED** [review]
👍

**Pull request merged**